btcutil/p2a_address_test.go

```diff
@@ -0,0 +1,155 @@
+package btcutil_test
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg"
+)
+
+// TestAddressPayToAnchor tests the AddressPayToAnchor type.
+func TestAddressPayToAnchor(t *testing.T) {
+	tests := []struct {
+		name        string
+		net         *chaincfg.Params
+		wantAddress string
+	}{
+		{
+			name:        "mainnet",
+			net:         &chaincfg.MainNetParams,
+			wantAddress: "bc1pfeessrawgf",
+		},
+		{
+			name:        "testnet",
+			net:         &chaincfg.TestNet3Params,
+			wantAddress: "tb1pfees9rn5nz",
+		},
+		{
+			name:        "regtest",
+			net:         &chaincfg.RegressionNetParams,
+			wantAddress: "bcrt1pfeesnyr2tx",
+		},
+		{
+			name:        "simnet",
+			net:         &chaincfg.SimNetParams,
+			wantAddress: "sb1pfeesxv0pfa",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			addr, err := btcutil.NewAddressPayToAnchor(tt.net)
+			if err != nil {
+				t.Fatalf("NewAddressPayToAnchor() error = %v",
+					err)
+			}
+
+			got := addr.EncodeAddress()
+			if got != tt.wantAddress {
+				t.Errorf("EncodeAddress() = %v, want %v",
+					got, tt.wantAddress)
+			}
+
+			if addr.String() != tt.wantAddress {
+				t.Errorf("String() = %v, want %v",
+					addr.String(), tt.wantAddress)
+			}
+
+			if !addr.IsForNet(tt.net) {
+				t.Errorf("IsForNet() = false, want true")
+			}
+
+			otherNet := &chaincfg.MainNetParams
+			if tt.net == &chaincfg.MainNetParams {
+				otherNet = &chaincfg.TestNet3Params
+			}
+			if addr.IsForNet(otherNet) {
+				t.Errorf("IsForNet(other) = true, want false")
+			}
+
+			expectedScript := []byte{0x51, 0x02, 0x4e, 0x73}
```

Comment:
I think it's better if I we declare something like the declaration in txscript/standard.go

```
var PayToAnchorScript = []byte{OP_1, OP_DATA_2, 0x4e, 0x73}
```